request.go

```diff
@@ -474,6 +498,10 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
 					continue
 				}
 
+				if pFieldType, ok := polyrelationFields[args[1]]; ok && fieldValue.Type() != pFieldType {
+					continue
+				}
+
```

Comment:
I think the solution makes sense but is not at all obvious to the casual reader. Would you mind adding a comment about what this achieves?

```suggestion
                                 // Skipping nonmatching types when a polymorphic relation of this name exists enables
                                 // a migration path from relation to polyrelation.
				if pFieldType, ok := polyrelationFields[args[1]]; ok && fieldValue.Type() != pFieldType {
					continue
				}

```